### PR TITLE
Merge with junit-parse

### PR DESCRIPTION
**Pull request by janisz**
This PR merges junit2jira with https://github.com/stackrox/junit-parse

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -119,6 +125,36 @@ func run(p params) error {
 //go:embed htmlOutput.html.tpl
 var htmlOutputTemplate string
 
+func (j junit2jira) convertToSlack(tc []testCase) error {
```
Comment (porridge) [inline main.go:128]: :thinking:
```suggestion
func (j junit2jira) maybeConvertToSlack(tc []testCase) error {
```

File: main.go [hunk 2]
```diff
@@ -570,3 +607,131 @@ func truncateSummary(s string) string {
...
+
+		attachments = append(attachments, failureAttachment)
+
+		// We've reached the desired message limit. We need to break out of all the loops
+		if len(attachments) > 3 {
+			break
+		}
+	}
+
+	if len(failedTestsBlocks) <= 0 {
```
Comment (porridge) [inline main.go:641]:
```suggestion
	if len(failedTestsBlocks) == 0 {
```

File: main.go [hunk 3]
```diff
@@ -570,3 +607,131 @@ func truncateSummary(s string) string {
...
+func failureToAttachment(title string, tc testCase) (slack.Attachment, error) {
+
+	failureMessage := tc.Message
+	failureValue := tc.Error
+	if tc.Error == tc.Message {
+		failureValue = ""
+	}
+
+	if failureMessage == "" && failureValue == "" {
+		return slack.Attachment{}, fmt.Errorf("no junit failure message or value for %s", title)
```
Comment (porridge) [inline main.go:669]:
```suggestion
		return slack.Attachment{}, fmt.Errorf("no junit failure message or error for %s", title)
```

File: main.go [hunk 4]
```diff
@@ -119,6 +125,36 @@ func run(p params) error {
 //go:embed htmlOutput.html.tpl
 var htmlOutputTemplate string
 
+func (j junit2jira) convertToSlack(tc []testCase) error {
+	if j.slackOutput == "" {
+		return nil
+	}
+	slackMsg := convertJunitToSlack(tc...)
+	if slackMsg == nil {
+		log.Printf("warning: no slack message set")
```
Comment (porridge) [inline main.go:134]: As a user I wouldn't know what this means. But do we really need to warn about apparently successful cases?

**Review by porridge — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (janisz) [reply main.go:128]: Why `maybe`? I've never seen this prefix for a function that returns error?

File: main.go (hunk 4 above)
Comment (janisz) [reply main.go:134]: good point this left from the original code

File: main.go (hunk 1 above)
Comment (porridge) [reply main.go:128]: Because this function is a no-op if `j.slackOutput == ""`. Or perhaps move the condition to the call site and then the name is fine.

Comment (janisz) [reply main.go:128]: let's name it `createSlackMessage` and have same convention as with other functions

https://github.com/stackrox/junit2jira/blob/d36b86edd823b074a7dbde0bc7a1f4b4620f0866/main.go#L157-L160

https://github.com/stackrox/junit2jira/blob/d36b86edd823b074a7dbde0bc7a1f4b4620f0866/main.go#L193-L196

**Pull request merged**